editor/Editor.cpp

```diff
@@ -48,18 +51,17 @@ static inline void addResourceFolderSelected() {
 
 class ModelViewerPanel : public IPanel {
 public:
-    ModelViewerPanel() : IPanel(TR("ui.window.title"), true) {
+    ModelViewerPanel() : IPanel(TR("ui.engineview"), true) {
         this->flags |=
-                ImGuiWindowFlags_NoTitleBar   |
-                ImGuiWindowFlags_NoDecoration |
-                ImGuiWindowFlags_NoResize     |
-                ImGuiWindowFlags_NoBackground ;
+                ImGuiWindowFlags_NoBackground;
     }
 
+    // Opens a file dialog used to select a model definition
     void addModelSelected() {
-        std::string path = Dialogs::openResource("*.json");
-        if (!path.empty())
-            this->setLoadedFile(path);
+        std::string path = FilesystemResourceProvider::getResourceIdentifier(modeldialog.GetSelected().string());
+        if (path.empty())
+            return Dialogs::popupError(TR("error.modelviewer.file_is_not_resource"));
+        this->setLoadedFile(path);
```

Comment:
Regression from main, the code that was replaced fixed a bug

Reply:
whoops! I must have forgotten to update that part! Will fix that in a moment!